### ISSUE TITLE
Add support for `Platform.OS`

### ISSUE THREAD

**Issue by okwasniewski** [callstack/react-native-visionos#20]
### Description

The goal of this issue is to add support for `Platform.OS = 'visionos'`. 

The work might require writing a custom metro resolver.

**Comment by thymikee** [comment]
Update from my work in this area: 
Managed to add support for `visionos` as a new platform (similar to `macos`). It revealed that we either need to convert/bridge existing `.ios.js` files to `visionos.js` or add a fallback platform. Tried the fallback platform solution by customizing `reactNativePlatformResolver`:
https://github.com/callstack/react-native-visionos/blob/ca6ac0a461579f4e6dc0781150541c66aa048ef1/packages/community-cli-plugin/src/utils/metroPlatformResolver.js#L28-L44
to accept
```ts
platformImplementations: {
  [platform: string]: [string, string | undefined],
}
```
as an argument, where the tuple would read as `[platform = "visionos", basePlatform = "ios"]`. This way I tricked Metro into resolving ios extensions as fallbacks, however this revealed another issue with `Platform.OS` being `visionos`, while internally we have lots of `Platform.OS === "ios"` calls. Which we would either need to augment with `visionos` checks (as tried here by @okwasniewski: https://github.com/callstack/react-native-visionos/pull/2/commits/e5b83e0dc7be12dd889ad5eaf7b39d74d8d3a17f) or entirely replace. Which adds maintenance overhead for an out-of-tree platform like this. Even then, for React Native VisionOS, which largely shares the same code with React Native iOS, it would mean we wouldn't be able to use most of the community iOS native packages, unless contributing extra platform checks. That's not scalable.

As a potential followup, I'm going to explore extending `Platform` primitive with a new "variant" concept. Variant would serve as an extension for the base platform. `visionos` would be variant of `ios`. `tvos` would be variant of `ios`. Maybe even `macos` could be variant of `ios`, but that's debatable. Adding variant support would likely mean either hacking around Platform concept between Metro and RN for iOS. Or extending Metro with this concept.
